lib/setupmanagers/qemuhck/storage_manager.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require_relative '../../auxiliary/json_helper'
+require_relative '../../auxiliary/host_helper'
+require_relative '../../auxiliary/string_helper'
+
+# AutoHCK module
+module AutoHCK
+  # QemuMachine class
+  class QemuMachine
+    # StorageManager class
+    class StorageManager
+      include Helper
+
+      IMAGE_FORMAT = 'qcow2'
+
+      def initialize(id, client_id, qemu_config, qemu_options, logger)
+        @id = id
+        @client_id = client_id
+        @logger = logger
+
+        @workspace_path = qemu_options['workspace_path']
+        @qemu_img_bin = qemu_config['qemu_img_bin']
+        @fs_test_image = qemu_config['fs_test_image']
+
+        @boot_image_path = Pathname.new(qemu_config['images_path']).join(qemu_options['image_name'])
+        @test_image_path = Pathname.new(@workspace_path).join("client#{@client_id}_test_image.#{IMAGE_FORMAT}")
+      end
+
+      def read_device(device)
+        @logger.info("Loading device: #{device}")
+        device_json = "#{QemuMachine::DEVICES_JSON_DIR}/#{device}.json"
+        unless File.exist?(device_json)
+          @logger.fatal("#{device} does not exist")
+          raise(InvalidConfigFile, "#{device} does not exist")
+        end
```

Comment:
You should check if `Json.read_json` raises the "file not found" error.

Generally, it's a bad practice to check a file's property before opening the file as it can change between the check and file opening. We do not really care such a race condition in this particular case, but it's better to always stick to the best practice.

I also wonder if this error is really necessary. Perhaps it may be fine to just let `Json.read_json` raise an error.

Reply:
Fixed